searchlib/src/main/java/com/yahoo/searchlib/expression/FloatBucketResultNode.java

```diff
@@ -88,13 +88,25 @@ protected int onCmp(ResultNode rhs) {
         FloatBucketResultNode b = (FloatBucketResultNode)rhs;
         double f1 = from;
         double f2 = b.from;
+        if (Double.isNaN(f1)) {
+            if (Double.isNaN(f2)) {
+                return 0;
+            }
+            return -1;
+        }
+        if (Double.isNaN(f2)) {
+            return 1;
+        }
         if (f1 < f2) {
             return -1;
         } else if (f1 > f2) {
             return 1;
         } else {
             double t1 = to;
             double t2 = b.to;
+            if (Double.isNaN(t2)) {
+                return 1;
+            }
```

Comment:
`t1` and `t2` are not equal even when `.to` both is `NaN`?

Reply:
right, needs fix